### PR TITLE
Update access-modifiers.md

### LINKED ISSUE #27195: `private protected` – does not allow `internal` access
> private protected: The type or member can be accessed only within its declaring assembly, by code in the same class or in a type that is derived from that class.

Maybe I just misunderstood the interpunction here, so it is meant like:

_The type or member can be accessed only within its declaring assembly: by code in the same class or in a type that is derived from that class._

From [private protected (C# Reference)](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private-protected)
> A private protected member is accessible by types derived from the containing class, but only within its containing assembly. 

That is correct definition. `private protected` allows _visibility_ to derived and base type within same assembly.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8121f526-cfe4-091d-d031-2cdc274cd8ee
* Version Independent ID: 76a389f6-b568-3568-6483-6df440f96f0c
* Content: [Access Modifiers - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers)
* Content Source: [docs/csharp/programming-guide/classes-and-structs/access-modifiers.md](https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/classes-and-structs/access-modifiers.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by Naidile-P-N**
Update description for private-protected

Fixes #27195

### CODE REVIEW

File: docs/csharp/programming-guide/classes-and-structs/access-modifiers.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ All types and type members have an accessibility level. The accessibility level
 - [protected](../../language-reference/keywords/protected.md): The type or member can be accessed only by code in the same `class`, or in a `class` that is derived from that `class`.
 - [internal](../../language-reference/keywords/internal.md): The type or member can be accessed by any code in the same assembly, but not from another assembly.
 - [protected internal](../../language-reference/keywords/protected-internal.md): The type or member can be accessed by any code in the assembly in which it's declared, or from within a derived `class` in another assembly.
-- [private protected](../../language-reference/keywords/private-protected.md): The type or member can be accessed only within its declaring assembly, by code in the same `class` or in a type that is derived from that `class`.
+- [private protected](../../language-reference/keywords/private-protected.md): The type or member can be accessed by types derived from the containing `class`, but only within its containing assembly.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/access-modifiers.md:19]: I think this wording might be clearer.  What do you think:

```suggestion
- [private protected](../../language-reference/keywords/private-protected.md): The type or member can be accessed by types derived from the `class` and declared within its containing assembly.
```

**Review by BillWagner — COMMENTED** [review]
Thanks for doing this @Naidile-P-N 

I had one suggestion to consider for wording before merging. Let me know what you think.

File: docs/csharp/programming-guide/classes-and-structs/access-modifiers.md (hunk 1 above)
Comment (Naidile-P-N) [reply docs/csharp/programming-guide/classes-and-structs/access-modifiers.md:19]: @BillWagner Can we make it - 

`The type or member can be accessed by types derived from the `class` that are declared within its containing assembly.`

**Review by BillWagner — APPROVED** [review]
Thanks for submitting this and discussing the wording. I've committed the suggestion, and I'll :shipit: now.

**Comment by Uzivatel919** [comment]
@BillWagner, I can tell I did not find this wording wise.

Consider protected modifier
> [protected](../../language-reference/keywords/protected.md): The type or member can be accessed only by code in the same `class`, or in a `class` that is derived from that `class`.

It enumerates both options: _declaring class_ and _derived class_ (in unconstrained space).

While
> [private protected](../../language-reference/keywords/private-protected.md): The type or member can be accessed by types derived from the `class` that are declared within its containing assembly.

enumerates just _derived class_ option.

Though it is just inconsistance – declaring class accessibility is implied by virtue of class encapsulation – still can be found confusing.

**Pull request merged**